packages/calypso-e2e/src/lib/blocks/audio-block.ts

```diff
@@ -0,0 +1,50 @@
+import { Page, ElementHandle } from 'playwright';
+
+const selectors = {
+	block: '.wp-block-audio',
+	fileInput: '.components-form-file-upload input[type="file"]',
+	spinner: '.components-spinner',
+};
+
+/**
+ * Represents the Audio block.
+ */
+export class AudioBlock {
+	static blockName = 'Audio';
+	block: ElementHandle;
+
+	/**
+	 * Constructs an instance of this block.
+	 *
+	 * @param {ElementHandle} block Handle referencing the block as inserted on the Gutenberg editor.
+	 */
+	constructor( block: ElementHandle ) {
+		this.block = block;
+	}
+
+	/**
+	 * Uplaods the target file at the supplied path to WPCOM.
+	 *
+	 * @param {string} path Path to the file on disk.
+	 */
+	async upload( path: string ): Promise< void > {
+		// The `input` element has a display:none style set and it cannot be found normally even if
+		// waitForSelector({state: 'hidden'}) is used.
+		const input = ( await this.block.$( selectors.fileInput ) ) as ElementHandle;
```

Comment:
As explained in the comments, normally a `waitForSelector()` would be used here.

However, even with `waitForSelector({state: 'hidden'})` the call would inexplicably hang.
Since the `GutenbergEditorComponent.addBlock` waits for the block to be fully inserted and stable, it was decided that simply querying for the hidden `input` selector in the block was acceptable.

Reply:
Just curious if `waitForSelector( { state: 'attached' } )` would work here? It is unnecessary as you mentioned because we're querying from a block that's already been rendered, but was just wondering if `display: none` has any impact on this state.

Reply:
It seems to work locally, so I've updated all three files to use `attached`! Thanks for the suggestion!